baseapp/baseapp.go

```diff
@@ -60,6 +60,7 @@ type BaseApp struct { // nolint: maligned
 	txDecoder         sdk.TxDecoder // unmarshal []byte into sdk.Tx
 
 	anteHandler    sdk.AnteHandler  // ante handler for fee and auth
+	postHandler    sdk.AnteHandler  // post handler, optional, e.g. for tips
```

Comment:
I went for the exact same signature for posthandlers as antehandlers.

We could also change a bit, e.g. having it manipulate the `*sdk.Result`:

```go
PostHandle(ctx sdk.Context, res *sdk.Result, tx sdk.Tx, simulate bool, next sdk.PostHandler) (sdk.Context,  res *sdk.Result, error)
```

Any thoughts?